hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/addressbook/AddressBookServiceImpl.java

```
@@ -0,0 +1,243 @@
+import com.hedera.mirror.importer.repository.AddressBookRepository;
+import com.hedera.mirror.importer.repository.FileDataRepository;
+import com.hedera.mirror.importer.util.Utility;
+
+@Log4j2
+@Named
+@AllArgsConstructor
+public class AddressBookServiceImpl implements AddressBookService {
+    public static final EntityId ADDRESS_BOOK_101_ENTITY_ID = EntityId.of(0, 0, 101, EntityTypeEnum.FILE);
+    public static final EntityId ADDRESS_BOOK_102_ENTITY_ID = EntityId.of(0, 0, 102, EntityTypeEnum.FILE);
+
+    private final AddressBookRepository addressBookRepository;
+    private final FileDataRepository fileDataRepository;
+
+    @Override
+    public void update(FileData fileData) {
+        if (!isAddressBook(fileData.getEntityId())) {
+            log.warn("Not an address book File ID. Skipping processing ...");
+            return;
+        }
+
+        if (fileData.getFileData() == null || fileData.getFileData().length == 0) {
+            log.warn("Byte array contents were empty. Skipping processing ...");
+            return;
+        }
+
+        try {
+            parse(fileData);
+        } catch (Exception e) {
+            log.error("Unable to parse address book", e);
+        }
+    }
+
+    @Override
+    public AddressBook getCurrent() {
+        Instant now = Instant.now();
+        long consensus_timestamp = Utility.convertToNanos(Instant.now().getEpochSecond(), now.getNano());
+        Optional<AddressBook> optionalAddressBook = addressBookRepository
+                .findLatestAddressBook(consensus_timestamp, ADDRESS_BOOK_102_ENTITY_ID.getId());
+
+        if (optionalAddressBook.isPresent()) {
+            AddressBook addressBook = optionalAddressBook.get();
+            log.info("Loaded addressBook from {} with nodes ({}).", addressBook.getStartConsensusTimestamp(),
+                    addressBook.getNodeSet());
+            return addressBook;
+        }
+
+        log.warn("No addressBooks before {} were found.", consensus_timestamp);
```

I don't think any current user of this method can handle a null returned object, so we'll get NPEs. Better to throw a meaningful exception as it's truly an invalid state if we don't have at least one address book. Thus, most of this method can be collapsed to
```java
return addressBookRepository.findLatestAddressBook(consensus_timestamp, ADDRESS_BOOK_102_ENTITY_ID.getId()).orElseThrow(() -> new IllegalStateException(""));
```

---

Funny had as a throw when I started too. Took your simplification also.